k4GaudiPandora/src/DDCaloDigi.cc

```diff
@@ -1181,409 +1119,411 @@ float DDCaloDigi::ahcalEnergyDigi(float energy, int id) {
+//                          << std::endl;
+//     return 0;
+//   }
 
-        for (int i = 0; i < 3; i++) {
-          streamlog_out(DEBUG) << htt.getPosition()[i] << " ";
-        }
-        streamlog_out(DEBUG) << endl;
+//   //CellIDDecoder<SimCalorimeterHit> idDecoder( col );
+//   // This variable is for tracking detectors only, not for calorimeters
+//   // std::string initString = m_geoSvc->constantAsString(m_encodingStringVariable.value());
 
-        // std::cout << "K-1 = " << km1 ;
-        // std::cout << "; GEARlay = " << gearlayer;
-        // std::cout << "; m = " << m;
-        // std::cout << "; s = " << s;
-        // std::cout << "; i = " << i_index;
-        // std::cout << "; j = " << j_index << std::endl;
+//   // get the encoding string from the original collection
+//   const auto initStringStrip = cellIDHandle.get();
+//   dd4hep::DDSegmentation::BitFieldCoder bitFieldCoder(initStringStrip);
 
-        assert(0);
-      }
+//   // make the new output collection
+//   //edm4hep::MutableSimCalorimeterHit tempstripcol = inputCaloHitCollection.create();;
+//   auto stripcol = edm4hep::SimCalorimeterHitCollection();
+//   //FIXME stripcol.setFlag(_flag.getFlag());
+//   //FIXME: add initString/ cellIDEncoding to stripcol
 
-      // set it to the existing one
-      newhit = htt;
-
-    } else {  // it's the first hit from this strip
-      // create a new hit for the strip
-      newhit = &stripcol.create();
-      ;
-      newhits[cellid] = newhit;
-      newHit.setCellID(cellID);
-      newhit.setPosition(stripPos);
-      newhit.setEnergy(0);  // this is added when we add MC contributions
-    }
+//   // link between strip CellID and hits storage for strip hits
+//   std::map<int, edm4hep::MutableSimCalorimeterHit*> newhits;  //SORT
 
-    // add the MC contriutions
-    float eadd(0);
-    auto  singleHit = hit.getContributions();
-
-    for (int ij = 0; ij < singleHit.size(); ij++) {
-      auto contrib = stripcol.create();
-      contrib.setParticle(hit.getParticle(ij));
-      contrib.setEnergy(hit.getEnergy(ij) * energyNonuniformityScaling);
-      contrib.setTime(hit.getTime(ij));
-      contrib.setPDG(hit.getPDG(ij));
-      newhit.addToContributions(contrib);
-      eadd += hit.getEnergy(ij) * energyNonuniformityScaling;
-    }
+//   // loop over input collection
+//   int numElements = col.size();
 
-    float esum(0);
-    auto  singlenewHit = newhit.getContributions();
-    for (int ij = 0; ij < singlenewHit.size(); ij++) {
-      esum += newhit.getEnergy(ij);
-    }
-  }  // loop over hits
+//   // // sum energy for check
+//   float tempenergysum(0);
+//   for (const auto& hit : col) {
+//     tempenergysum += hit.getEnergy();
+//   }
 
-  // move the hits from the temporary storage to the output collection
-  for (std::map<int, edm4hep::MutableSimCalorimeterHit*>::iterator itt = newhits.begin(); itt != newhits.end(); itt++) {
-    stripcol->addElement(itt->second);
-  }
+//   float scTVirtLengthBar(-99);
```

Comment:
Can all this stuff that is commented out be removed?

Reply:
So the whole function `combineVirtualStripCells` is not needed?